airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -443,3 +446,18 @@ def streams(self, empty_schema: bool = False) -> Iterable:
                     "properties": self._stream_properties(fp, empty_schema=empty_schema, read_sample_chunk=True),
                 }
         yield AirbyteStream(name=self.stream_name, json_schema=json_schema, supported_sync_modes=[SyncMode.full_refresh])
+
+    def openpyxl_chunk_reader(self, file):
+        """Use openpyxl lazy loading feature to read excel files in chunks of 500 lines at a time"""
+        work_book = load_workbook(filename=file, read_only=True)
+        for sheetname in work_book.sheetnames:
+            work_sheet = work_book[sheetname]
+            data = work_sheet.values
+            cols = next(data)
```

Comment:
empty generator can kill you by StopIteration, is it ok ?

Reply:
updated;